flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -1772,9 +1773,23 @@ private StateBackend loadStateBackend(ReadableConfig configuration, ClassLoader
                         + ContinuousFileMonitoringFunction.MIN_MONITORING_INTERVAL
                         + " ms.");
 
+        int readerParallelism = getParallelism();
+
+        boolean isAdaptiveBatchSchedulerEnabled =
+                readerParallelism == ExecutionConfig.PARALLELISM_DEFAULT
+                        && getConfig().isDynamicGraph();
+
+        if (isAdaptiveBatchSchedulerEnabled) {
+            readerParallelism =
+                    getConfiguration()
```

Comment:
`getConfiguration()` may be empty when executing sql jobs and will make this fix ineffective in that case. This problem seems to be caused by `DummyStreamExecutionEnvironment` which does not properly override `getConfiguration()`. But we are still looking into some details to figure out whether it is safe to fix it by simply do the overriding.

Reply:
yes, I found this problem too. I agree to keep this PR open before the problem is fixed.